src/gc/gc.d

```diff
@@ -3132,19 +3134,17 @@ struct SmallObjectPool
 
                 rt_finalizeFromGC(q, size, attr);
 
-                toClear |= GCBits.BITS_1 << clearIndex;
+                freeBits = true;
+                toFree.set(i);
 
                 debug(COLLECT_PRINTF) printf("\tcollecting %p\n", p);
                 //log_free(sentinel_add(p));
 
                 debug (MEMSTOMP) memset(p, 0xF3, size);
-                freebits.set(biti);
```

Comment:
Think we should not free the objects as it could otherwise lead to undeterministic memory corruptions.
https://trello.com/c/InwzqvHZ/7-runfinalizers-shouldn-t-free-objects-or-needs-to-be-renamed